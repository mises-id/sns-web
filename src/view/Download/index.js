/*
 * @Author: lmk
 * @Date: 2021-07-16 00:15:24
 * @LastEditTime: 2022-10-27 13:59:25
 * @LastEditors: lmk
 * @Description: Airdrop page
 */
import { getReferralUrl } from "@/api/user";
import { isIosPlatform } from "@/utils";
import React, { useEffect, useState } from "react";
import "./index.scss";
const Download = () => {
  const defaultUrl = isIosPlatform() ? "https://testflight.apple.com/join/dFykgML6" : "https://play.google.com/store/apps/details?id=site.mises.browser";
  const [downloadUrl, setdownloadUrl] = useState(defaultUrl)
  useEffect(() => {
    const referrer = sessionStorage.getItem('referrer')
    if(referrer){
      const misesId = referrer.indexOf('did:mises:')>-1 ? referrer.split('did:mises:')[1] : referrer
      getDownloadUrl('landing', misesId)
    } else {
      const search = new URLSearchParams(window.location.search);
      const misesId = search.get('misesid');
      getDownloadUrl('invite', misesId)
    }
  }, [])

  const getDownloadUrl = (medium, misesId)=>{
    if(misesId){
      getReferralUrl({
        misesId,
        type:'url',
        medium
      }).then(res=>{
        const url = isIosPlatform() ? res.ios_medium_link : res.medium_url
        setdownloadUrl(url)
      })
    }
  }
  return (
    <>
      <div>
        <div className="mises-bg" style={{backgroundImage:"url('/static/images/bg.png')"}}>
          <div className="mises-logo-wrap">
            <img src="/static/images/download-logo.png" alt="" className="download-logo"/>
            <img src="/static/images/MisesBrowser.png" alt=""  className="download-name"/>
            <p>The world's first extension-supported </p>
            <p>mobile Web3 browser</p>
          </div>
        </div>
      </div>
      <img src="/static/images/download.png" alt="" className="download-png"/>
      <a href={downloadUrl} target="_blank" className="download" rel="noreferrer">Download</a>
    </>
  );
};
export default Download;
