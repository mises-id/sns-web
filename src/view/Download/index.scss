.mises-bg{
  background-size: 100% 90px;
  background-repeat: no-repeat;
  display: flex;
}
.download-logo{
  width: 60px;
  display: inline-block;
}
.mises-txt{
  font-size: 30px;
  font-weight: bold;
  color: #333333;
  display: inline-block;
  margin-left: 10px;
}
.download-png{
  width: calc(100% - 100px);
  display: block;
  margin: 25px auto;
}
.option{
  font-size: 14px;
  font-family: SF UI Display;
  font-weight: 400;
  color: #333333;
  line-height: 25px;
  margin:0 27px;
  p{
    word-break: break-word;
  }
}
.download{
  width: calc(100vw - 80px);
  height: 50px;
  background: #5D61FF;
  border-radius: 25px;
  margin: 40px auto 25px;
  display: block;
  text-align: center;
  line-height: 50px;
  color: #FFFFFF;
  font-size: 17px;
  font-family: SF UI Display;
  font-weight: bold;
  color: #FFFFFF;
}
.mises-logo-wrap{
  width: 100vw;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  .download-logo{
    width: 85px;
    margin-top: 30px;
  }
  .download-name{
    width: 233px;
    margin-top: 20px;
    margin-bottom: 20px;
  }
  p{
    font-size: 14px;
    line-height: 18px;
  }
}