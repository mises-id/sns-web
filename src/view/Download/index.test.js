/*
 * @Author: lmk
 * @Date: 2021-07-16 10:05:53
 * @LastEditTime: 2022-03-31 09:49:39
 * @LastEditors: lmk
 * @Description: Download test
 */

import React from 'react';
import {shallow } from 'enzyme';
import Download from '.';
describe('GreatePosts', () => {
  it('should render correctly', () => {
    const component = shallow(< Download/>)
    expect(component).toMatchSnapshot();
  });
});