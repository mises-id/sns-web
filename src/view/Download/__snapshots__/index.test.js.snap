// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreatePosts should render correctly 1`] = `
<Fragment>
  <div>
    <div
      className="mises-bg"
      style={
        Object {
          "backgroundImage": "url('/static/images/bg.png')",
        }
      }
    >
      <div
        className="mises-logo-wrap"
      >
        <img
          alt=""
          className="download-logo"
          src="/static/images/download-logo.png"
        />
        <img
          alt=""
          className="download-name"
          src="/static/images/MisesBrowser.png"
        />
        <p>
          The world's first extension-supported 
        </p>
        <p>
          mobile Web3 browser
        </p>
      </div>
    </div>
  </div>
  <img
    alt=""
    className="download-png"
    src="/static/images/download.png"
  />
  <a
    className="download"
    href="https://play.google.com/store/apps/details?id=site.mises.browser"
    rel="noreferrer"
    target="_blank"
  >
    Download
  </a>
</Fragment>
`;
